### PR TITLE
Use models for validation

### PR DESCRIPTION
**Pull request by Netmisa**
# Description

This Pr use models for validation.
Body and path changed

- [x] `/v1/patches/location` for `StopPoint` with tests
- [x] `/v1/patches/location` for `Elevator` with tests
- [x] `/v1/patches/location/from_reporter` for `StopPoint` with tests
- [x] `/v1/elevators/import/csv` with tests

Documentation updated: [Swagger UI](http://petstore.swagger.io/?url=https://raw.githubusercontent.com/Netmisa/Hermod/code_update/swagger-spec.yaml)

### CODE REVIEW

File: src/AppBundle/Model/Api/Json/LocationPatch/Base.php [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+ *     map = {
+ *      "stop_point": "AppBundle\Model\Api\Json\LocationPatch\StopPoint",
+ *      "elevator": "AppBundle\Model\Api\Json\LocationPatch\Elevator"
+ *     })
+ */
+class Base
+{
+    /**
+     * @Assert\NotBlank(message="This field is missing.")
+     * @Assert\Regex("/^stop_point|elevator/")
```
Comment (vincentlepot) [inline src/AppBundle/Model/Api/Json/LocationPatch/Base.php]: Using Assert\Choice would be more readable?

File: src/AppBundle/Model/Api/Json/LocationPatch/Gps.php [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+use JMS\Serializer\Annotation as JMS;
+use AppBundle\Validator as AppAssert;
+use AppBundle\Model\Api\Json\LocationPatch\Location;
+
+class Gps
+{
+    /**
+     * @Assert\NotNull()
+     * @Assert\Type("float")
+     * @JMS\Type("float")
```
Comment (vincentlepot) [inline src/AppBundle/Model/Api/Json/LocationPatch/Gps.php]: + only positive values ?

File: src/AppBundle/Services/LocationPatch.php [hunk 3]
```diff
@@ -0,0 +1,84 @@
...
+    private $em;
+    private $equipmentEntity;
+
+    function __construct(EntityManagerInterface $em)
+    {
+        $this->em = $em;
+        $this->equipmentEntity = null;
+    }
+
+    private function createLocation(User $user, $data, bool $withUserLocation)
```
Comment (vincentlepot) [inline src/AppBundle/Services/LocationPatch.php]: Naming: User and withUserLocation, we are not talking about the same "User". withReporterLocation ?

File: src/AppBundle/Services/LocationPatch.php [hunk 4]
```diff
@@ -0,0 +1,84 @@
...
+            ->findOneBy([
+                'code' => $elevator->getCode(),
+                'sourceName' => $elevator->getSource()->getName()
+            ])
+        ;
+
+        return $elevatorEntity;
+    }
+
+    public function create(User $user, $data, bool $withUserLocation = false)
```
Comment (vincentlepot) [inline src/AppBundle/Services/LocationPatch.php]: Naming: same as createLocation

File: src/AppBundle/Model/Api/Json/LocationPatch/Gps.php (hunk 2 above)
Comment (Netmisa) [reply src/AppBundle/Model/Api/Json/LocationPatch/Gps.php]: Yes, we see together last time for `accuracy` type. We hesitate between `int` or `float`

File: src/AppBundle/Controller/v1/LocationPatchController.php [hunk 5]
```diff
@@ -30,7 +25,7 @@ public function getAllAction()
     {
         return new JsonResponse(
             $this->getDoctrine()
-                ->getRepository('AppBundle:LocationPatch')
+                ->getRepository('AppBundle:Location')
```
[the remark this answers is not in the document]
Comment (Netmisa) [reply src/AppBundle/Controller/v1/LocationPatchController.php]: If you want, but we maybe do the same for `Status` => `StatusPatch`

File: tests/AppBundle/Controller/v1/ApiTestCase.php [hunk 6]
```diff
@@ -96,6 +91,10 @@ protected function assertError(ResponseInterface $response, $statusCode, $errorT
 
     protected function assertResourceCreated(ResponseInterface $response)
     {
+        if ($response->getStatusCode() == 500){
+//            dump($response->getBody()->getContents());
```
Comment (vincentlepot) [inline tests/AppBundle/Controller/v1/ApiTestCase.php]: To be removed

File: tests/AppBundle/Controller/v1/ApiTestCase.php [hunk 7]
```diff
@@ -96,6 +91,10 @@ protected function assertError(ResponseInterface $response, $statusCode, $errorT
...
+            die;
```
Comment (vincentlepot) [inline tests/AppBundle/Controller/v1/ApiTestCase.php]: Hmm.. Are you sure? An exception perhaps?

Comment (Netmisa) [reply tests/AppBundle/Controller/v1/ApiTestCase.php]: My bad

File: src/AppBundle/Model/Api/Json/LocationPatch/Gps.php (hunk 2 above)
Comment (Netmisa) [reply src/AppBundle/Model/Api/Json/LocationPatch/Gps.php]: See together we will change to `int` type

File: src/AppBundle/Controller/v1/LocationPatchController.php (hunk 5 above)
[the remark this answers is not in the document]
Comment (Netmisa) [reply src/AppBundle/Controller/v1/LocationPatchController.php]: See together I will rename `Status` to `StatusPatch` and `Location` to `LocationPatch`

**Pull request merged**